jetstream/external_config.py

```diff
@@ -39,6 +45,133 @@ def validate(self, experiment: jetstream.experimenter.Experiment) -> None:
         Analysis("no project", "no dataset", conf).validate()
 
 
+def check_all_keys_lowercase(config: MutableMapping[str, Any]) -> List[str]:
+    """
+    Recursively check that configuration keys are lowercase.
+
+        Returns a dictionary with a dict key path to key which is invalid.
+    """
+
+    invalid_keys = list()
+    for key, value in config.items():
+        if isinstance(value, dict):
+            invalid_keys.extend(check_all_keys_lowercase(value))
+
+        if key.lower() != key:
+            only_uppercase_letters = list(filter(str.isupper, key))
+            if (
+                "".join(sorted(set(only_uppercase_letters), key=only_uppercase_letters.index))
+                == "GB"
+            ):
+                # handling rare occasion where uppercase is used:
+                # example: https://github.com/mozilla/jetstream-config/blob/main/bug-1722551-pref-full-js-parsing-experiment-nightly-94-94.toml#L803  # noqa: E501
+                continue
+
+            invalid_keys.append(key)
```

Comment:
Mh, maybe instead of handling every special case here, we should just make sure that `config` can handle differences in casing? For example, by lowercasing all the keywords and identifiers when parsing the config.